Tasks/ServiceFabricDeploy/Tests/WindowsAuthDeploy.ps1

```diff
@@ -0,0 +1,62 @@
+[CmdletBinding()]
+param()
+
+. $PSScriptRoot\..\..\..\Tests\lib\Initialize-Test.ps1
+
+$publishProfilePath = "$PSScriptRoot\data\NoAuthPublishProfile.xml"
+$applicationPackagePath = "random package path"
+$serviceConnectionName = "random connection name"
+$serviceFabricSdkModulePath = "$PSScriptRoot\data\ServiceFabricSDK.ps1"
+$appName = "AppName"
+$overwriteBehavior = "SameAppTypeAndVersion"
+
+# Setup input arguments
+Register-Mock Get-VstsInput { $publishProfilePath } -- -Name publishProfilePath
+Register-Mock Get-VstsInput { $applicationPackagePath } -- -Name applicationPackagePath -Require
+Register-Mock Get-VstsInput { $serviceConnectionName } -- -Name serviceConnectionName -Require
+Register-Mock Get-VstsInput { "false" } -- -Name compressPackage
+Register-Mock Get-VstsInput { $overwriteBehavior } -- -Name overwriteBehavior
+Register-Mock Get-VstsInput { "false" } -- -Name skipUpgradeSameTypeAndVersion
+Register-Mock Get-VstsInput { "false" } -- -Name skipPackageValidation
+Register-Mock Get-VstsInput { "false" } -- -Name unregisterUnusedVersions
+
+# Setup file resolution
+Register-Mock Find-VstsFiles { $publishProfilePath } -- -LegacyPattern $publishProfilePath
+Register-Mock Find-VstsFiles { $applicationPackagePath } -- -LegacyPattern $applicationPackagePath -IncludeDirectories
+
+Register-Mock Assert-VstsPath
+Register-Mock Test-Path { $true } -- "HKLM:\SOFTWARE\Microsoft\Service Fabric SDK"
+
+# Setup mock VSTS service endpoint
+$vstsEndpoint = @{
+    "Auth" = @{
+        "Scheme" = "None"
+        "Parameters" = @{
+            "IsWindowsCredential" = "true"
+        }
+    }
+}
+Register-Mock Get-VstsEndpoint { $vstsEndpoint } -- -Name $serviceConnectionName -Require
+
+# Setup mock for connection to cluster
+Register-Mock Connect-ServiceFabricCluster { $null } -- -ConnectionEndpoint "test" -WindowsCredential
```

Comment:
Please add an Assert-WasCalled for the Mock so that you can be sure Connect is getting called with -WindowsCredential